parsers/abc.c

```diff
@@ -0,0 +1,85 @@
+/*
+*   Copyright (c) 2009, Eric Forgeot
+*
+*   Based on work by Jon Strait
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your opinion) any later version.
+*
+*   This module contains functions for generating tags for Abc files
+*   (https://en.wikipedia.org/wiki/ABC_notation).
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"	/* must always come first */
+
+#include <ctype.h>
+#include <string.h>
+
+#include "parse.h"
+#include "read.h"
+#include "vstring.h"
+#include "routines.h"
+#include "entry.h"
+
+/*
+*   DATA DEFINITIONS
+*/
+
+typedef enum {
+   K_SECTION,
+} AbcKind;
+
+static kindDefinition AbcKinds[] = {
+	{ true, 's', "section", "sections" },
+};
+
+/*
+*   FUNCTION DEFINITIONS
+*/
+
+static void makeAbcTag (const vString* const name)
+{
+	tagEntryInfo e;
+	initTagEntry (&e, vStringValue(name), K_SECTION);
+
+	makeTagEntry(&e);
+}
+
+static void findAbcTags (void)
+{
+	vString *name = vStringNew();
+	const unsigned char *line;
+
+	while ((line = readLineFromInputFile()) != NULL)
+	{
+		if (line[0] == 'T') {
+			vStringCatS(name, " / ");
+			vStringCatS(name, (const char *) line);
+			makeAbcTag(name);
```

Comment:
You can use makeSimpleTag declared in "parse.h" instead of defining a parser specific helper function.